Work bench PR for one-go Core Data extraction into target

mokagio opened the pull request:
Part of #24165

mokagio committed:
Convert WordPressData from Swift package to Xcode target

To much Objective-C code using logic implemented elsewhere in Swift for
us to be able to move the files in two separate Objective-C- and Swift-
only packages.

Refer to https://github.com/wordpress-mobile/WordPress-iOS/pull/24166 as
an experiment for this that shows the various build failure that even a
partial attempt at moving some files results in.

mokagio committed:
Disable automatic code signing for WordPressData framework

mokagio committed:
Disable "Build libraries for distribution" flag in WordPressData

WordPressAuthenticator does not use it either

mokagio committed:
Let WordPressData read iOS deployment target from the project config

mokagio committed:
Use the same `SWIFT_INSTALL_OBJC_HEADER` as WordPressAuthenticator

mokagio committed:
Link WordPressData to the Jetpack target

🤦‍♂️

mokagio committed:
Add CocoaLumberjack to the WordPressData dependencies

mokagio committed:
Move `NSManagedObject+Lookup` and `TaggedManagedObjectID` to WordPressData

mokagio committed:
Disable `wpAssertionFailure` in `CoreDataHelper`

mokagio committed:
Add WordPressKit as a dependency to WordPressData

mokagio committed:
Move `PostContentProvider.h` to WordPressData

mokagio committed:
Move `PostVisibility` definition in `AbstractPost.swift`

Make it easier to extract it in the future.

mokagio committed:
Move various file in staging folder before move to WordPressData

Just a convenience so it's easier to find them later.

mokagio committed:
Move some models and their deps to WordPressData – Build fails

mokagio committed:
Move a few more files, the next will be `WPAccount`...

mokagio committed:
Move WPAccount and related dependencies - Build is worse than before

mokagio committed:
Remove a stray trailing new line from `PromptRemindersScheduler.swift`

mokagio committed:
Move yet more files...

mokagio committed:
Add SFHFKeychainUtils as a WordPressData dependency

mokagio committed:
Remove a couple of `WordPress-Swift.h` imports from WordPressData

mokagio committed:
Do not treat warnings as errors in WordPressData

There's a deprecation warning that we can't yet get rid of...

mokagio committed:
Move more files — Can't find WordPressData-Swift.h?!

mokagio committed:
Add `NSURL+IDN` as a WordPressData dependency

mokagio committed:
Add CocoaLumberjack (not the Swift) as a WordPressData dependency

mokagio committed:
Add a bunch of missing imports

The only way I could reveal these, was by commenting out a log of other
code that was giving issues because written in Swift but used in
Objective-C.

mokagio committed:
Move `defaultWordPressComAccountRestAPI` definition to WordPressData

mokagio committed:
Remove knowledge of `AppConfiguration` from WPAccount Objective-C

mokagio committed:
Add precompiled header to WordPressData to configure CocoaLumberjack

mokagio committed:
Remove documentation header params without descripiton

The tighter compiler settings in WordPressData reveal them with
warnings.

mokagio committed:
Add some missing imports and convert to angle bracket imports

mokagio committed:
Add dedicated `.m` file for setting `DDLogLevel`

mokagio committed:
With these hacks, WordPressData compiles as a standalone framework

But the unit tests don't because... it builds the
WordPressShareExtension?!

mokagio committed:
Fix issue with WordPressData tests attempting to build WordPress

It had remained there as a target dependency...

The tests now pass :)

mokagio committed:
Disable CocoaLumberjack temporarily in WordPressData

mokagio committed:
Start addressing build failures when building WordPress

Now that WordPressData builds, too...

mokagio committed:
Address some more import and access control issues to use WordPressData

mokagio committed:
Move `Domain.swift` to WordPressData

mokagio committed:
Move `Role` to WordPressData

mokagio committed:
Access control update in `Page`

mokagio committed:
Move `BlogQuery` to WordPressData

mokagio committed:
Access control updates

mokagio committed:
Move `Array+Page.swift` to WordPressData

mokagio committed:
Move `Blog+Quota` to WordPressData

mokagio committed:
Add CI step for fast iteration tests on WordPressData

mokagio committed:
Remove more incomplete documentation comments

The tighter compiler in WordPressData flagged them with warnings.

mokagio committed:
Address some more access control build failures

mokagio committed:
Add more necessary import and access control changes

mokagio committed:
Some more imports and access control changes

dangermattic commented:
<!--
  0 Errors
  3 Warnings: <code>Modules/Package.swift</c...
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><code>Modules/Package.swift</code> was changed without updating its corresponding <code>Package.resolved</code>. Please resolve the Swift packages as appropriate to your project setup (e.g. in Xcode or by running <code>swift package resolve</code>).</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Move `PublicizeInfo` to WordPressData

mokagio committed:
Move `SiteSuggestion` to WordPressData

mokagio committed:
Move `UserSuggestion` to WordPressData

mokagio committed:
More access control changes to get the code to compile

mokagio committed:
More access control changes

mokagio committed:
Update more access control levels

mokagio committed:
Move `PageTemplateCategory` to WordPressData

mokagio committed:
Move `PageTemplateLayout` to WordPressData

mokagio committed:
More imports and access control

mokagio committed:
Actually add `UserSettings` to the WordPressData

mokagio committed:
More...

mokagio committed:
Remove `WPAccount.h` imports outside WordPressData

mokagio committed:
Remove imports of classes that are now in WordPressData

mokagio committed:
More...

mokagio committed:
Rebase - Move Comment to WordPressData

mokagio committed:
Remove unused `AccountService.h` import

mokagio committed:
Move `ReaderPost` and related dependencies

mokagio committed:
More...

mokagio committed:
Address a couple for `FIXME`s from the `ReaderPost` move

mokagio committed:
More...

mokagio committed:
Update `DataMigrator` for older `SharedDataIssueSolver` changes

Only noticed the build failure now because of Xcode only showing a few
build errors at a time...

mokagio committed:
More...

mokagio committed:
Remove redundant `public` access-level

mokagio committed:
Remove useless `NSURL+IDN` import

Unfortunately, not enough to remove the dependency

mokagio committed:
Begin to address test build failures...

mokagio committed:
Move WordPressData files from staging folder to target folder

mokagio committed:
Split WordPressData in Objective-C and Swift folders

mokagio committed:
Use same module settings as WordPressAuthenticator

mokagio committed:
Make WordPress unit tests compile

mokagio committed:
Use `xcconfig` for WordPressData, same as WordPressAuthenticator

mokagio committed:
Use default `PRODUCT_MODULE_NAME` definition

mokagio committed:
Add WordPressData tests to the main unit tests `xctestplan`

mokagio committed:
Add `BlockedAuthor` to WordPressData

mokagio committed:
Move Gutenberg models to WordPressData

mokagio committed:
Add FormattableContentKit dependency to WordPressData

mokagio committed:
Move more models, build still broken

mokagio committed:
Address SwiftLint violations

mokagio committed:
Add some missing imports

mokagio committed:
Add more files to WordPressData target

- `BlockedAuthor`
- `BlockedSite`
- `DiffAbstractValue`
- `DiffContentValue`
- `DiffTitleValue`
- `GravatarProfile`
- `InviteLinks`
- `Plan`
- `PlanFeature`
- `PlanGroup`
- `PostCategory+Creation`
- `QuickStartTourState`
- `ReaderSearchSuggestion`
- `Revision`
- `RevisionDiff`
- `SharingButton`
- `SiteInfo`
- `SiteInformation`
- `Suggestion`
- `ThemeIdHelper`
- `UserProfile`

mokagio committed:
Move other files to WordPressData

- `BlockEditorSettings`
- `BlockEditorSettingElement`
- `LikeUser`
- `LikeUserPreferredBlog`

mokagio committed:
Move most of the Blogging Prompts models to WordPressData

Required gutting out some bits of it to avoid having to deal with more
dependencies for the time being

mokagio committed:
Move more models to WordPressData

- `Menu`
- `MenuItem`
- `MenuLocation`
- `PostTag`
- `PostType`

mokagio committed:
Make WordPress and Jetpack targets build

mokagio committed:
Make WordPress unit tests build

mokagio committed:
Add Gravatar as a dependency to WordPressData

mokagio committed:
Remove unused WordPressUI import from `ManagedPerson`

mokagio committed:
Move `ManagedPerson` to WordPressData

mokagio committed:
Remove `ReachabilityUtils.onAvailableInternetConnectionDo`

mokagio committed:
Move `Notification` to WordPressData

mokagio committed:
Fix `WPAccount+Keychain` location in WordPressData subfolders

mokagio committed:
Move `WordPress.xcdatamodeld` to WordPressData

mokagio committed:
Move `Blog` `BlockEditorSettings` `Capabilities` and `Plans` extensions

mokagio committed:
Convert a `Blog support...` from Objective-C to Swift

mokagio committed:
Port `Blog` `dotComID` and `willSave` to Swift

mokagio committed:
Port `siteVisibility` `Blog` property to Swift

mokagio committed:
Update after rebase - SharingAuthorizationHelper.m

mokagio committed:
Update after rebase - Blog.m

mokagio committed:
Update after rebase - WPAccount+RestApi.swift

mokagio committed:
Fix after rebase - PostVisibility.swift

mokagio committed:
Remove duplicated import

mokagio committed:
Fix after rebase - StatsViewController.m

mokagio committed:
Remove code disable during WordPressData move in `Comment`

mokagio committed:
Remove code disable during WordPressData move in `ReaderSiteTopic`

mokagio committed:
Re-enable `wpAssert` calls in WordPressData thanks to #24346

mokagio commented on WordPress/WordPressData/Swift/Blog+Capabilities.swift:
```diff
@@ -76,11 +76,20 @@ extension Blog {
     public func areBloggingRemindersAllowed() -> Bool {
-        return isUserCapableOf(.EditPosts) && JetpackNotificationMigrationService.shared.shouldPresentNotifications()
+        // FIXME:
+        return isUserCapableOf(.EditPosts) /* && JetpackNotificationMigrationService.shared.shouldPresentNotifications() */
```
Addressed by https://github.com/wordpress-mobile/WordPress-iOS/pull/24353

mokagio commented on WordPress/WordPressData/Swift/AbstractPost.swift:
```diff
@@ -10,7 +11,8 @@ extension AbstractPost {
     var isNewDraft: Bool {
-        wpAssert(isOriginal(), "Must be called on the original")
+        // FIXME: Delaying wpAssert migration because of underlying dependencies
+        // wpAssert(isOriginal(), "Must be called on the original")
```
This and all the others are addressed by https://github.com/wordpress-mobile/WordPress-iOS/pull/24346

mokagio commented on WordPress/WordPressData/Swift/Blog.swift:
```diff
@@ -0,0 +1,239 @@
+        case .stats:
+            return supportsRestApi() && isViewingStatsAllowed()
+        case .stockPhotos:
+            // FIXME: JetpackFeaturesRemovalCoordinator not available yet
+//            return supportsRestApi() && JetpackFeaturesRemovalCoordinator.jetpackFeaturesEnabled()
+            return supportsRestApi()
+        case .tenor:
+            // FIXME: JetpackFeaturesRemovalCoordinator not available yet
+//            return JetpackFeaturesRemovalCoordinator.jetpackFeaturesEnabled()
```
Addressed in e6b2d962463293e0f10f31b7833e340e5884f0bf

mokagio commented on WordPress/WordPressData/Swift/Blog.swift:
```diff
@@ -0,0 +1,239 @@
+        case .stats:
+            return supportsRestApi() && isViewingStatsAllowed()
+        case .stockPhotos:
+            // FIXME: JetpackFeaturesRemovalCoordinator not available yet
+//            return supportsRestApi() && JetpackFeaturesRemovalCoordinator.jetpackFeaturesEnabled()
+            return supportsRestApi()
+        case .tenor:
+            // FIXME: JetpackFeaturesRemovalCoordinator not available yet
+//            return JetpackFeaturesRemovalCoordinator.jetpackFeaturesEnabled()
+            fatalError()
```
Addressed e6b2d962463293e0f10f31b7833e340e5884f0bf

mokagio committed:
Remove code disable during WordPressData move in `ReaderCard`

mokagio commented on Modules/Sources/FormattableContentKit/NotificationContentFactory.swift:
```diff
@@ -6,6 +6,7 @@ private enum Constants {
     static let Ranges = "ranges"
 }
 
+// FIXME: This is a WordPressData notification!
```
This should be fixable by moving some bits back to the app target. I might have been too quick in moving types...

mokagio replied:
Actually... I'm not sure it's a WordPressData Notification... Anyway, I got rid of `Notification` here altogether in https://github.com/wordpress-mobile/WordPress-iOS/pull/24354

mokagio committed:
Remove "module WordPress" from `xcdatamodel` entities that had it

This way, the app will look for the model object in WordPressData, which
is where they are defined.

mokagio committed:
Fix after rebase - Rename .enterprise.xcconfig to .alpha.xcconfig

mokagio committed:
Put all models in the global namespace

mokagio committed:
Fix after rebase - Remove duplicated `SiteOrganizationType`

mokagio committed:
Delete stray empty `NotificationContentFactory.swift` file

mokagio committed:
Remove `NotificationSyncMediatorProtocol` from WordPressData

Does not need to be in there.

mokagio committed:
Remove code disable during WordPressData move in `ReaderCrossPostMeta`

mokagio committed:
Update after rebase - Use `Common.*.xcconfig` for WordPressData

mokagio committed:
DRY test logic to generate `BlogSettings` instance

mokagio committed:
Address typecast warning for a `Blog` assignment in tests

mokagio committed:
Hack tests to avoid failing to get BlogSettings from DB

mokagio committed:
Restore do not treat warnings as errors after removing xcconfig

mokagio commented:
Closing in favor of https://github.com/wordpress-mobile/WordPress-iOS/pull/24378
